baseband/dada/base.py

```diff
@@ -430,19 +430,31 @@ def open(name, mode='rs', subset=None, header=None, **kwargs):
     is_template = isinstance(name, six.string_types) and ('{' in name and
                                                           '}' in name)
     is_sequence = isinstance(name, (tuple, list))
+    is_sequentialfile = isinstance(name, (sf.SequentialFileReader,
+                                          sf.SequentialFileWriter))
 
     if 'b' not in mode:
         if header is None:
             if 'w' in mode:
+                # If sequentialfile object, check that it's opened for writing.
+                if is_sequentialfile:
+                    assert name.mode == 'w+b', (
+                        'open only accepts sequential files opened in '
+                        '\'w+b\' mode for writing.')
                 # For writing a header is required.
-                if 'nthread' in kwargs:
-                    kwargs.setdefault('npol', kwargs.pop('nthread'))
                 header = DADAHeader.fromvalues(**kwargs)
                 kwargs = {}
 
+            elif is_sequentialfile:
```

Comment:
As I spent a minute or so being confused about which branch this was in, maybe just:
```
else:  # for readingadjusted
    if is_sequential_file:
...
```

Reply:
This doesn't work since `is_template and 'OBS_OFFSET' in name` is checked simultaneously, and leads to a TypeError.

Reply:
Ugh, I'm being silly.  Adding brackets to that line works fine.